wallet-new/src/Cardano/Wallet/WalletLayer/Error.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+-- | Types describing runtime errors related to
+-- wallet layers. It should be a common interface for
+-- all the errors popping up from the @WalletLayer@.
+
+module Cardano.Wallet.WalletLayer.Error
+    ( WalletLayerError (..)
+    ) where
+
+import           Universum
+
+import qualified Data.Text.Buildable
+import           Formatting (bprint, stext, (%))
+
+import           Cardano.Wallet.API.V1.Types (WalletId)
+
+
+data WalletLayerError
+    = MissingWallet WalletId
+    | MissingAccount
+    deriving (Show, Eq, Generic)
```

Comment:
I think I would prefer to see a finer-grained error type. If I do `getWallet :: WalletId -> m (Either WalletLayerError Wallet)`, then I have a ton of cases that won't ever arise.